### ISSUE TITLE
[Candidate 3v3 Aeon of Excalibur ZH v5] AI is unable to build tech tree until it reached Supply Docks very far away

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsRankedMaps#489]
## Observed behavior

AI is unable to build tech tree until it reached Supply Docks very far away.

## Expected behavior

AI can build its tech tree early without problems.

## Screenshots

![shot_20231125_173708_18](https://github.com/TheSuperHackers/GeneralsRankedMaps/assets/4720891/1a26c017-e81d-4c37-974d-87b168608c7f)

![shot_20231125_173712_19](https://github.com/TheSuperHackers/GeneralsRankedMaps/assets/4720891/38ceef88-e112-4795-ad07-0fab140e1ac2)

**Comment by xezon** [comment]
When playing 1v1v1v1v1v1 then the center AI will not build a Power Plant and anything else, perhaps because it cannot find a Supply Dock.

![shot_20231125_175611_21](https://github.com/TheSuperHackers/GeneralsRankedMaps/assets/4720891/11c0fe25-bb09-4d31-aefd-3a1fa4361e6a)

**Comment by toirehantaa** [comment]
it's a 3v3 map. Idk why you are even trying a 1v1v1v1v1. AI players can't build near enemies. They are too close. That the AI only builds near supply docks and not crates is due to the hardcoding of its behaviours. Not a map problem. Close the blocker please

**Comment by xezon** [comment]
Perhaps place an empty Supply Dock under the terrain to trick the AI into building Supply Depots?

**Comment by toirehantaa** [comment]
I fear that this will completely break the map :,)
Take a look at my map called "Urban Downfall" - I implemented a new thing called stacked supply docks. You can still collect from the top one technically even though you have a supply at the bottom. I suspect that for money collection, the height doesn't matter. So the AI and perhaps even the player will be able to collect from a supply stuck below the ground

**Comment by toirehantaa** [comment]
I also suggest you to not flag AI problems, except for when they don't work at all. Because they are always related to the AI itself and not to the map, sadly.

**Comment by xezon** [comment]
The map design must consider technical limitations and try to find ways to avoid running into problems.

To give example, if a certain asset placed in a map was crashing the game client, then the right course of action would be to not use that asset, instead of blaming the asset of being faulty.

**Comment by TimberOwl** [comment]
then the map design are gonna be same boring designs

**Comment by toirehantaa** [comment]
![image](https://github.com/TheSuperHackers/GeneralsRankedMaps/assets/115837820/fb426c45-078a-4695-bf59-ab28ad3dc707)
![image](https://github.com/TheSuperHackers/GeneralsRankedMaps/assets/115837820/600a5fd9-acce-40e9-aac5-1192a1f27733)
![image](https://github.com/TheSuperHackers/GeneralsRankedMaps/assets/115837820/ef38fa41-ae1b-4fe3-b5f0-11ba345934e1)
I'd be willing to change the crates into supply docks under the condition that they are sunken into the ground like this - as the supporting platform does not fit the base layout the slightest, further than this is something I don't want to do

**Comment by toirehantaa** [comment]
this would be fixing the AI building problem, so there's that

**Comment by xezon** [comment]
I see that space is tight. Supply Dock looks odd when sunken and clipping into mountain. Would it perhaps be possible to elevate the Docks over the mountains and water by placing some structures underneath that would setup the foundation for it?

**Comment by Jundiyy** [comment]
If the map is made for 3v3 then AI should be considered that way, it won't build properly if it has an enemy next to it, especially early on.
In this case, the testing should be done as teams and not Free For All. 

AI will only gather from x amount of money from Crates or Docks, otherwise will ignore making a Supply there, it's been a while but I think it was $10,500.

**Comment by xezon** [comment]
AI test was done in 3v3 and 1v1v1v1v1v1. This report is mainly concerned about 3v3 AI setup.

**Comment by Jundiyy** [comment]
I can't see any Docks in the base. If that's the case then the AI will not build up in the base. 
As mentioned above, there is a minimum amount required on the Docks or Crates before the AI will go there. You'd need to have one of the crates with a lot more money in the Base. 
If the problem is due to the Docks then I would remove the blocker, perhaps manual scripts could help here, never tried it though.
If a map has no Docks at all then I've seen the AI build near Crates.

**Comment by xezon** [comment]
Making one crate high value is also an option if it solves this issue. It would violate on of the Ranked Maps rules but having better functioning AI would be preferable I believe.

**Comment by TimberOwl** [comment]
kauf dir strick und nen stuhl

**Comment by toirehantaa** [comment]
I wont be editing small supply crates for the sake of having AI work on the map - since it's against the rules and since they give no optical feedback that they are having a higher value. It already is a problem for AI to be so close to each other in a team, the building space is rather limited for them anyways. You guys will have to accept that this map is not well suited for AI. As mentioned above, I can offer to put supplydocks into the ground like this if you really want AI to work properly, but this is as far as I am ready to go. I wont let the hang from the cliff like said before, it does not suit the style. So we will have docks like this or we will go back to crates - you decide

**Comment by xezon** [comment]
Yes changing value of Supply Piles is not optimal either. Either way it would violate current rules. Perhaps someone else has a clever idea.

Who is in favor of leaving AI as is? AI needs to drive along half the map before it can start building tech tree, so in practice it will be very easy to beat AI on this map by simply intercepting the worker unit on the side lanes.

**Comment by xezon** [comment]
Tanso has requested to demote this from blocker. Demoted.
